x-pack/plugins/case/server/common/models/commentable_case.ts

```diff
@@ -0,0 +1,300 @@
+        ? [{ type: subCaseSOType, name: `associated-${subCaseSOType}`, id: this.subCase.id }]
+        : []),
+    ];
+  }
+
+  private async update({ date, user }: { date: string; user: User }): Promise<CommentableCase> {
+    let updatedSubCaseAttributes: SavedObject<SubCaseAttributes> | undefined;
+
+    if (this.subCase) {
+      const updatedSubCase = await this.service.patchSubCase({
+        client: this.soClient,
+        subCaseId: this.subCase.id,
+        updatedAttributes: {
+          updated_at: date,
+          updated_by: {
+            ...user,
+          },
+        },
+        version: this.subCase.version,
+      });
+
+      updatedSubCaseAttributes = {
+        ...this.subCase,
+        attributes: {
+          ...this.subCase.attributes,
+          ...updatedSubCase.attributes,
+        },
+        version: updatedSubCase.version ?? this.subCase.version,
+      };
+    }
+
+    const updatedCase = await this.service.patchCase({
+      client: this.soClient,
+      caseId: this.collection.id,
+      updatedAttributes: {
+        updated_at: date,
+        updated_by: { ...user },
+      },
+      version: this.collection.version,
+    });
+
+    // this will contain the updated sub case information if the sub case was defined initially
+    return new CommentableCase({
+      collection: {
+        ...this.collection,
+        attributes: {
+          ...this.collection.attributes,
+          ...updatedCase.attributes,
+        },
+        version: updatedCase.version ?? this.collection.version,
+      },
+      subCase: updatedSubCaseAttributes,
+      soClient: this.soClient,
+      service: this.service,
+    });
+  }
+
+  /**
+   * Update a comment and update the corresponding case's update_at and updated_by fields.
+   */
+  public async updateComment({
+    updateRequest,
+    updatedAt,
+    user,
+  }: {
+    updateRequest: CommentPatchRequest;
+    updatedAt: string;
+    user: User;
+  }): Promise<UpdateCommentResp> {
+    const { id, version, ...queryRestAttributes } = updateRequest;
+
+    const [comment, commentableCase] = await Promise.all([
+      this.service.patchComment({
+        client: this.soClient,
+        commentId: id,
+        updatedAttributes: {
+          ...queryRestAttributes,
+          updated_at: updatedAt,
+          updated_by: user,
+        },
+        version,
+      }),
+      this.update({ date: updatedAt, user }),
+    ]);
+    return {
+      comment,
+      commentableCase,
+    };
+  }
+
+  /**
+   * Create a new comment on the appropriate case. This updates the case's updated_at and updated_by fields.
+   */
+  public async createComment({
+    createdDate,
+    user,
+    commentReq,
+  }: {
+    createdDate: string;
+    user: User;
+    commentReq: CommentRequest;
+  }): Promise<NewCommentResp> {
+    if (commentReq.type === CommentType.alert) {
+      if (this.status === CaseStatuses.closed) {
+        throw Boom.badRequest('Alert cannot be attached to a closed case');
```

Comment:
Ignore my previous comment about the removed check 😄

Reply:
haha